cmd/paths.go

```diff
@@ -256,77 +257,100 @@ func pathsFetchCmd(a *appState) *cobra.Command {
+			//note: path file names on the chain-registry are in alphabetical order ex: achain-zchain.json
+			p := permute.Slice(chains)
+			var chainCombinations []string
+			for p.Permute() {
+				tmp := make([]string, 2)
+				copy(tmp, chains)
+				sort.Strings(tmp)
+				tmp1 := string(tmp[0] + "-" + tmp[1])
+				if !contains(chainCombinations, tmp1) {
+					chainCombinations = append(chainCombinations, tmp1)
+				}
 			}
 
-			// Try to fetch path info for each configured chain that has canonical chain/path info in the GH repo
-			for _, srcChain := range a.Config.Chains {
-				for _, dstChain := range a.Config.Chains {
+			overwrite, _ := cmd.Flags().GetBool(flagOverwriteConfig)
 
-					// Add paths to rly config from {localRepo}/interchain/chaind-id/
-					localPathsDir := path.Join(localRepo, "interchain", srcChain.ChainID())
+			client := github.NewClient(nil)
 
-					dir := path.Clean(localPathsDir)
-					files, err := ioutil.ReadDir(dir)
-					if err != nil {
-						fmt.Fprintf(cmd.ErrOrStderr(), "path info does not exist for chain: %s. Consider adding its info to %s. Error: %v\n", srcChain.ChainID(), path.Join(PATHSURL, "interchain"), err)
-						break
-					}
+			for _, pthName := range chainCombinations {
+				_, exist := a.Config.Paths[pthName]
+				if exist && !overwrite {
+					fmt.Fprintf(cmd.ErrOrStderr(), "skipping:  %s already exists in config, use -o to overwrite (clears filters)\n", pthName)
+					continue
+				}
 
-					// For each path file, check that the dst is also a configured chain in the relayers config
-					for _, f := range files {
-						pth := filepath.Join(dir, f.Name())
-
-						if f.IsDir() {
-							fmt.Fprintf(cmd.ErrOrStderr(), "directory at %s, skipping...\n", pth)
-							continue
-						}
-
-						byt, err := os.ReadFile(pth)
-						if err != nil {
-							cleanupDir(localRepo)
-							return fmt.Errorf("failed to read file %s: %w", pth, err)
-						}
-
-						p := &relayer.Path{}
-						if err = json.Unmarshal(byt, p); err != nil {
-							cleanupDir(localRepo)
-							return fmt.Errorf("failed to unmarshal file %s: %w", pth, err)
-						}
-
-						if p.Dst.ChainID == dstChain.ChainID() {
-							pthName := strings.Split(f.Name(), ".")[0]
-							if err = a.Config.AddPath(pthName, p); err != nil {
-								return fmt.Errorf("failed to add path %s: %w", pth, err)
-							}
-
-							fmt.Fprintf(cmd.ErrOrStderr(), "added path %s...\n", pthName)
-						}
+				// TODO: Don't use github api. Potentially use: https://github.com/eco-stake/cosmos-directory once they integrate IBC data into restAPI. This will avoid rate limits.
+				fileName := pthName + ".json"
+				regPath := path.Join("_IBC", fileName)
+				client, _, err := client.Repositories.DownloadContents(cmd.Context(), "cosmos", "chain-registry", regPath, nil)
+				if err != nil {
+					if errors.As(err, new(*github.RateLimitError)) {
+						return fmt.Errorf("github API rate limit reached while querying data ERR: %w", err)
```

Comment:
You could drop the `ERR:` part at the end of this error message so it's just `error message: %w`, this is the convention we have started to use.